packages/core/contracts/optimistic-assertor/implementation/sovereign-security-manager/SuperbondSovereignSecurityManager.sol

```diff
@@ -0,0 +1,129 @@
+// same claim should always have increasing bond amounts. Once a preconfigured superbond amount is reached, this SSM
+// automatically takes over the arbitration of assertions for all matching claims.
+contract SuperbondSovereignSecurityManager is BaseSovereignSecurityManager, Ownable {
+    struct ArbitrationResolution {
+        bool valueSet;
+        bool resolution;
+    }
+
+    struct ClaimBonding {
+        bool superBondReached;
+        IERC20 currency;
+        uint256 currentBondAmount;
+    }
+
+    // Before this Optimistic Assertor is set via setOptimisticAssertor all assertions will revert.
+    OptimisticAssertorInterface public optimisticAssertor;
+
+    // Address of linked requesting contract. Before this is set via setAssertingCaller all assertions will be blocked.
+    address public assertingCaller;
+
+    mapping(bytes32 => ArbitrationResolution) public arbitrationResolutions;
+
+    mapping(IERC20 => uint256) public superBonds; //Superbond amounts for each currency.
+
+    mapping(bytes32 => ClaimBonding) public claimBondings; // Track the bondings for each claim.
+
+    event AssertingCallerSet(address indexed assertingCaller);
+    event SuperBondAmountSet(IERC20 indexed currency, uint256 superBondAmount);
+    event SuperBondReached(bytes32 indexed claimId, IERC20 indexed currency);
+
+    function setOptimisticAssertor(address optimisticAssertorAddress) public onlyOwner {
+        require(optimisticAssertorAddress != address(0), "Invalid address");
+        optimisticAssertor = OptimisticAssertorInterface(optimisticAssertorAddress);
+    }
+
+    // Setting superBondAmount to 0 will block all assertions for that currency.
+    function setSuperBondAmount(IERC20 currency, uint256 superBondAmount) public onlyOwner {
+        superBonds[currency] = superBondAmount;
+        emit SuperBondAmountSet(currency, superBondAmount);
+    }
+
+    // Set the address of the contract that will be allowed to use Optimistic Assertor.
+    // This can only be set once. We do not set this at constructor just to allow for some flexibility in the ordering
+    // of how contracts are deployed.
+    function setAssertingCaller(address _assertingCaller) public onlyOwner {
+        require(_assertingCaller != address(0), "Invalid asserting caller");
+        require(assertingCaller == address(0), "Asserting caller already set");
+        assertingCaller = _assertingCaller;
+        emit AssertingCallerSet(_assertingCaller);
+    }
+
+    function setArbitrationResolution(
+        bytes32 identifier,
+        uint256 time,
+        bytes memory ancillaryData,
+        bool arbitrationResolution
+    ) public onlyOwner {
+        bytes32 requestId = keccak256(abi.encode(identifier, time, ancillaryData));
+        arbitrationResolutions[requestId] = ArbitrationResolution(true, arbitrationResolution);
+    }
+
+    function processAssertionPolicies(bytes32 assertionId) public override returns (AssertionPolicies memory) {
```

Comment:
I think that we should ideally keep the `getAssertionPolicies` non-state modifying (read function) and use the dispute callback to update the internal trackers in this contract. Ideally, `getAssertionPolicies` is static.